kdc/kerberos5.c

```diff
@@ -2223,6 +2223,33 @@ _kdc_as_rep(astgs_request_t r)
 	goto out;
     }
 
+    /*
+     * Select the best encryption type for the KDC without regard to
+     * the client since the client never needs to read that data.
+     */
+
+    ret = _kdc_get_preferred_key(r->context, config,
+				 r->server, r->sname,
+				 &setype, &skey);
+    if(ret)
+	goto out;
+
+    /* If server is not krbtgt, fetch local krbtgt key for signing authdata */
+    if (is_tgs) {
+	krbtgt_key = skey;
+    } else {
+	ret = get_local_tgs(r->context, config, r->server_princ->realm,
```

Comment:
Ah, @elric1 has repeatedly proposed that we have a little cache of TGS principal lookups, with a cache entry lifetime of a few seconds.  Just a note to self here.  This reminded me of that.